File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/CoreData.java
package de.bund.digitalservice.ris.caselaw.domain;

import de.bund.digitalservice.ris.caselaw.domain.court.Court;
import de.bund.digitalservice.ris.caselaw.domain.lookuptable.documenttype.DocumentType;
import jakarta.validation.constraints.PastOrPresent;
import java.time.LocalDate;
import java.time.Year;
import java.util.List;
import lombok.Builder;
import org.hibernate.validator.constraints.UniqueElements;

@Builder(toBuilder = true)
public record CoreData(
    @UniqueElements List<String> deviatingDocumentNumbers,
    @UniqueElements List<String> fileNumbers,
    @UniqueElements List<String> deviatingFileNumbers,
    Court court,
    @UniqueElements List<String> deviatingCourts,
    DocumentType documentType,
    Procedure procedure,
    List<String> previousProcedures,
    String ecli,
    String celexNumber,
    @UniqueElements List<String> deviatingEclis,
    String appraisalBody,
    @PastOrPresent LocalDate decisionDate,
    // If hasDeliveryDate is true, then decisionDate is actually a deliveryDate
    boolean hasDeliveryDate,
    @UniqueElements List<LocalDate> oralHearingDates,
    @UniqueElements List<LocalDate> deviatingDecisionDates,
    String legalEffect,
    List<String> inputTypes,
    DocumentationOffice documentationOffice,
    DocumentationOffice creatingDocOffice,
    @UniqueElements List<String> leadingDecisionNormReferences,
    List<@PastOrPresent Year> yearsOfDispute,
    boolean isResolved,
    @PastOrPresent LocalDate resolutionDate,
    List<Source> sources) {}

File: frontend/src/domain/coreData.ts
import { Court } from "@/domain/court"
import { CourtBranchLocation } from "@/domain/courtBranchLocation"
import DocumentationOffice from "@/domain/documentationOffice"
import { DocumentType } from "@/domain/documentType"
import { Procedure } from "@/domain/procedure"

import { Source } from "@/domain/source"

export type CoreData = {
  deviatingDocumentNumbers?: string[]
  fileNumbers?: string[]
  deviatingFileNumbers?: string[]
  court?: Court
  deviatingCourts?: string[]
  courtBranchLocation?: CourtBranchLocation
  documentType?: DocumentType
  procedure?: Procedure
  previousProcedures?: string[]
  ecli?: string
  celexNumber?: string
  deviatingEclis?: string[]
  appraisalBody?: string
  decisionDate?: string
  hasDeliveryDate?: boolean
  oralHearingDates?: string[]
  deviatingDecisionDates?: string[]
  legalEffect?: string
  inputTypes?: string[]
  documentationOffice?: DocumentationOffice
  creatingDocOffice?: DocumentationOffice
  yearsOfDispute?: number[]
  leadingDecisionNormReferences?: string[]
  sources?: Source[]
  isResolved?: boolean
  resolutionDate?: string
}

export const coreDataLabels: {
  [coreDataKeyKey in keyof Required<CoreData>]: string
} = {
  celexNumber: "CELEX-Nummer",
  hasDeliveryDate: "Zustellung an Verkündungs statt",
  oralHearingDates: "Datum der mündlichen Verhandlung",
  court: "Gericht",
  deviatingCourts: "Fehlerhaftes Gericht",
  courtBranchLocation: "Sitz der Außenstelle",
  fileNumbers: "Aktenzeichen",
  deviatingFileNumbers: "Abweichendes Aktenzeichen",
  deviatingDocumentNumbers: "Abweichende Dokumentnummer",
  decisionDate: "Entscheidungsdatum",
  appraisalBody: "Spruchkörper",
  creatingDocOffice: "Erstellende Dokumentationsstelle",
  ecli: "ECLI",
  documentationOffice: "Dokumentationsstelle",
  inputTypes: "Eingangsart",
  isResolved: "Erledigt",
  yearsOfDispute: "Streitjahr",
  resolutionDate: "Erledigungsmitteilung",
  sources: "Quelle",
  deviatingDecisionDates: "Abweichendes Entscheidungsdatum",
  documentType: "Dokumenttyp",
  deviatingEclis: "Abweichender ECLI",
  procedure: "Vorgang",
  previousProcedures: "Vorgang Historie",
  legalEffect: "Rechtskraft",
  leadingDecisionNormReferences: "BGH Nachschlagewerk",
}
